### ISSUE TITLE
Allow Collections To Be Nested

### ISSUE THREAD

**Issue by Undistraction** [netlify/netlify-cms#2056]
**Is your feature request related to a problem? Please describe.**
Currently all collections are displayed at the top level of the menu in the sidebar. This means that a collection like Pages is at the same level of hierachy as StaffMembers which might only be used on a single page. 

**Describe the solution you'd like**
I'd like control over nesting of Collections, so a page-specific Collection or Collections could be nested inside a group for that Page

**Describe alternatives you've considered**
There is no available alternative.

**Additional context**

I'd like the ability to have the sidebar look something like:

```
- Pages
- About Page
      - Staff List
      - About Slides
- Contact Page
      - Contact Details
```

**Comment by erquhart** [comment]
This is about the sidebar UI and not about underlying file placement, correct?

**Comment by Undistraction** [comment]
@erquhart That's correct. It feels like we should be able to group collections in the sidebar. This would have no effect on file location, only on the grouping of the collections in the sidebar. Even just a single level of grouping would be really helpful. So to make my above outline a little clearer:

```
- Pages [Collection]
- About Page [Group]
      - Staff List [Collection]
      - About Slides [Collection
- Contact Page [Group]
      - Contact Details [Collection]
```
The ideal would be able to group collections within other collection items, but I think that would be really complex:
```
- Pages [Collection]
   - About Page [Collection Item]
      - Staff List [Collection]
      - About Slides [Collection
  - Contact Page [Collection Item]
      - Contact Details [Collection]
```

The problem I'm trying to solve, is, lets say I need a component called 'Stats' that contains a list of stats that I need to managed from the CMS. That component and its data/collection will only ever be used on the Home page, but I have no way of associating that data/collection with the home page. A non-programmer who is using the CMS expects to be able to edit the content on the homepage from the Home collection item in the Pages Collection - this is where they edit the title of the page, its metadata, and the body copy etc, however they need to find the Stats collection and edit that to make a change on the home page. Currently the Stats collection lives at the root along with all the other collections, and this is very confusing for them.

**Comment by erquhart** [comment]
Yep, got it. The challenge here is that file collection items, like "About" and "Contact" are listed in the main area of the Collections view when you click their parent collection ("Pages"). With the model you're suggesting, this would no longer be the case and the entire collection would be listed in the sidebar.

I don't think this is a bad idea necessarily, as file collections are generally limited in size, but it does introduce one UI oddity: clicking a sidebar item in collections always opens a collection, but this change would introduce sidebar items that open a specific entry for editing.

I suppose if we made an obvious distinction between collections and entries in the sidebar, this would be fine.

All of that to say, it's a non-trivial change, but would definitely be nice, agreed.

Somewhat relevant: https://github.com/netlify/netlify-cms/issues/535

**Comment by Undistraction** [comment]
I think this is still a valid request.

**Comment by agrawalaayushi** [comment]
Any update on this feature?

**Comment by erezrokah** [comment]
Hi @agrawalaayushi, we're trying to work our way through the most request features first:
https://github.com/netlify/netlify-cms/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Upvoting this feature will help prioritize it.

**Comment by robsonsobral** [comment]
May I suggest something easier?

A property `parentLabel` to the collections. All collections with the same value should be shown together, under the same item on the sidebar.
